apollo-federation/src/sources/connect/json_selection/selection_set.rs

```diff
@@ -0,0 +1,421 @@
+//! Functions for applying a [`SelectionSet`] to a [`JSONSelection`]. This creates a new
+//! `JSONSelection` mapping to the fields on the selection set, and excluding parts of the
+//! original `JSONSelection` that are not needed by the selection set.
+
+use std::collections::HashSet;
+
+use apollo_compiler::executable::Selection;
+use apollo_compiler::executable::SelectionSet;
+
+use crate::sources::connect::json_selection::Alias;
+use crate::sources::connect::json_selection::NamedSelection;
+use crate::sources::connect::JSONSelection;
+use crate::sources::connect::Key;
+use crate::sources::connect::PathSelection;
+use crate::sources::connect::SubSelection;
+
+/// Find all fields in a selection set with a given name.
+fn find_fields_in_selection_set<'a>(set: &'a SelectionSet, name: &str) -> Vec<&'a Selection> {
+    let mut fields = vec![];
+    for selection in &set.selections {
+        match selection {
+            Selection::Field(field) => {
+                if field.name == name {
+                    fields.push(selection);
+                }
+            }
+            Selection::FragmentSpread(_) => {
+                // Ignore - these are always expanded into inline fragments
+            }
+            Selection::InlineFragment(fragment) => {
+                fields.extend(find_fields_in_selection_set(&fragment.selection_set, name));
+            }
+        }
+    }
+    fields
+}
+
+impl JSONSelection {
+    pub fn apply_selection_set(&self, selection_set: &SelectionSet) -> Self {
```

Comment:
Done

Reply:
Actually, @benjamn - @dylan-apollo pushed back on this trait since there's really no use for it, and I agree. We aren't using it as a trait bound or for dynamic dispatch, so it's kind of a useless abstraction. It basically just requires pulling that trait into scope to call the method. Would you be ok with removing the trait? Unless you see a way that I should be using this trait that I'm not currently?

Reply:
I went ahead and removed the trait for now. It can be added back if you see a need for it.